zebra-chain/src/parameters/network/tests/vectors.rs

```diff
@@ -0,0 +1,151 @@
+                    );
+                }
+            }
+        }
+
+        assert_eq!(
+            network.coin_type(),
+            zp_network.coin_type(),
+            "Parameters::coin_type() outputs must match"
+        );
+        assert_eq!(
+            network.hrp_sapling_extended_spending_key(),
+            zp_network.hrp_sapling_extended_spending_key(),
+            "Parameters::hrp_sapling_extended_spending_key() outputs must match"
+        );
+        assert_eq!(
+            network.hrp_sapling_extended_full_viewing_key(),
+            zp_network.hrp_sapling_extended_full_viewing_key(),
+            "Parameters::hrp_sapling_extended_full_viewing_key() outputs must match"
+        );
+        assert_eq!(
+            network.hrp_sapling_payment_address(),
+            zp_network.hrp_sapling_payment_address(),
+            "Parameters::hrp_sapling_payment_address() outputs must match"
+        );
+        assert_eq!(
+            network.b58_pubkey_address_prefix(),
+            zp_network.b58_pubkey_address_prefix(),
+            "Parameters::b58_pubkey_address_prefix() outputs must match"
+        );
+        assert_eq!(
+            network.b58_script_address_prefix(),
+            zp_network.b58_script_address_prefix(),
+            "Parameters::b58_script_address_prefix() outputs must match"
+        );
+    }
+}
+
+/// Checks that `NetworkUpgrade::activation_height()` returns the activation height of the next
+/// network upgrade if it doesn't find an activation height for a prior network upgrade, and that the
+/// `Genesis` upgrade is always at `Height(0)`.
+#[test]
+fn activates_network_upgrades_correctly() {
+    let expected_activation_height = 1;
+    let network = testnet::Parameters::build()
+        .activation_heights(ConfiguredActivationHeights {
+            nu5: Some(expected_activation_height),
+            ..Default::default()
+        })
+        .to_network();
+
+    let genesis_activation_height = NetworkUpgrade::Genesis
+        .activation_height(&network)
+        .expect("must return an activation height");
+
+    assert_eq!(
+        genesis_activation_height,
+        Height(0),
+        "activation height for all networks after Genesis and BeforeOverwinter should match NU5 activation height"
+    );
+
+    for nu in NETWORK_UPGRADES_IN_ORDER.into_iter().skip(1) {
+        let activation_height = nu
+            .activation_height(&network)
+            .expect("must return an activation height");
+
+        assert_eq!(
+            activation_height, Height(expected_activation_height),
+            "activation height for all networks after Genesis and BeforeOverwinter \
+            should match NU5 activation height, network_upgrade: {nu}, activation_height: {activation_height:?}"
+        );
+    }
+}
+
+/// Checks that there are no duplicate activation heights when using configured activation heights.
+// TODO: Convert this to a proptest.
```

Comment:
Actually, this seems unnecessary, it's really only here to make sure we don't remove the filter for `BeforeOverwinter` without updating `.split_off(&Height(2))` to `Height(1)`.

```suggestion
// TODO: Find out if it's okay for `BeforeOverwinter` to be activated at a later height, and remove this test if it is, 
//       see <https://github.com/ZcashFoundation/zebra/pull/8383/files#r1563215645>
```